### DISCUSSION TITLE
Any way to extract punctuation marks and transcription

### DISCUSSION THREAD

**Discussion by gormat** [snakers4/silero-models#42] [Q&A] [answered]
Hey. 
Really cool project !!!
I am using silero-stt model to extract text. 
I just have few questions:
- This model does not extract punctuation marks and does not split into sentences, am I missing something?
- Is there a way to get start and end timestamps for each word as well?

Thank you.

**Comment by snakers4** [comment, marked as answer]
Hi @gormat,

> This model does not extract punctuation marks and does not split into sentences, am I missing something?

in most of cases written punctuation is not related to how people actually speak (except for maybe `?`)
in the CE version of our models we just provide the STT part
but typically the pipeline may work as follows:

- you can use some [VAD](https://github.com/snakers4/silero-vad) to split speech into utterances. typically people separate ideas with pauses
- you can tune some pre-trained LM like a transformer to add capital letters, commas, full-stops etc

> Is there a way to get start and end timestamps for each word as well?

- Please use this colab [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/snakers4/silero-models/blob/master/examples.ipynb)
- Proceed to PyTorch example
- Search for comment `# align example`

**Comment by snakers4** [comment]
> you can tune some pre-trained LM like a transformer to add capital letters, commas, full-stops etc

We actually published models for this
